cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,307 @@
+     */
+    public static Map<String, String> generateMapWithApiCredentials(Map<String, UserEntry> entries) {
+        Map<String, String> data = new HashMap<>(3);
+        data.put(REBALANCE_OPERATOR_PASSWORD_KEY, Util.encodeToBase64(entries.get(REBALANCE_OPERATOR_USERNAME).password()));
+        data.put(HEALTHCHECK_PASSWORD_KEY, Util.encodeToBase64(entries.get(HEALTHCHECK_USERNAME).password()));
+        data.put(AUTH_FILE_KEY, Util.encodeToBase64(generateApiAuthFileAsString(entries)));
+        return data;
+    }
+
+    /**
+     * Returns Cruise Control API auth credentials file following Jetty's
+     * HashLoginService's file format: username: password [,rolename ...]
+     * as a String.
+     *
+     * @param entries Map of API user entries containing API user credentials.
+     *
+     * @return Returns Cruise Control API auth credentials file as a String.
+     */
+    private static String generateApiAuthFileAsString(Map<String, UserEntry> entries) {
+        StringBuilder sb = new StringBuilder();
+        for (UserEntry e : entries.values()) {
+            sb.append(e.username).append(": ").append(e.password).append(", ").append(e.role).append("\n");
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Generates a new API secret for Cruise Control by aggregating credentials from various sources.
+     * This method collects API credentials from three potential sources:
+     *   (1) Old Cruise Control API secret
+     *   (2) User-managed API secret
+     *   (3) Topic operator-managed API secret.
+     * It uses these credentials to create a comprehensive map of API credentials, which is then used to generate a new API secret
+     * for Cruise Control.
+     *
+     * @param passwordGenerator the password generator used for creating new credentials.
+     * @param oldCruiseControlApiSecret the existing centralized API secret, containing previously stored credentials.
+     * @param userManagedApiSecret the secret managed by the user, containing user-defined API credentials.
+     * @param topicOperatorManagedApiSecret the secret managed by the topic operator, containing credentials for the topic operator.
+     * @return a new Secret object containing the aggregated API credentials for Cruise Control.
+     */
+    public Secret generateApiSecret(PasswordGenerator passwordGenerator,
+                                    Secret oldCruiseControlApiSecret,
+                                    Secret userManagedApiSecret,
+                                    Secret topicOperatorManagedApiSecret) {
+
+        Map<String, ApiCredentials.UserEntry> apiCredentials = new HashMap<>();
+        apiCredentials.putAll(generateCoManagedApiCredentials(passwordGenerator, oldCruiseControlApiSecret));
+        apiCredentials.putAll(generateUserManagedApiCredentials(userManagedApiSecret, userManagedApiSecretKey));
+        apiCredentials.putAll(generateToManagedApiCredentials(topicOperatorManagedApiSecret));
```

Comment:
Would it make sense to pass the three parameters into `generateMapWithApiCredentials` instead of creating a map just to pass them there?

Reply:
The map as a whole is also used for the value of the AUTH_FILE_KEY entry in the map that is created in `generateMapWithApiCredentials` (There is a map turned to a String for a value in the map of `generateMapWithApiCredentials`). I could move the map creation there, but I still think it would be more simple to leave it where is

We would still need to create the same map in the method it is passed to

Reply:
> I could move the map creation there

I think that would be much better, because it would be much clearer what the parameters are and you don't really save anything by doing it outside.

Reply:
Just tried moving it over, you are right it does look neater